src/main/java/fi/metatavu/famifarm/reporting/xlsx/XlsxYieldReport.java

```diff
@@ -0,0 +1,243 @@
+package fi.metatavu.famifarm.reporting.xlsx;
+
+import java.io.OutputStream;
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import fi.metatavu.famifarm.batches.BatchController;
+import fi.metatavu.famifarm.events.EventController;
+import fi.metatavu.famifarm.events.HarvestEventController;
+import fi.metatavu.famifarm.events.SowingEventController;
+import fi.metatavu.famifarm.events.WastageEventController;
+import fi.metatavu.famifarm.localization.LocalesController;
+import fi.metatavu.famifarm.localization.LocalizedValueController;
+import fi.metatavu.famifarm.persistence.model.Batch;
+import fi.metatavu.famifarm.persistence.model.Event;
+import fi.metatavu.famifarm.persistence.model.HarvestEvent;
+import fi.metatavu.famifarm.persistence.model.Product;
+import fi.metatavu.famifarm.persistence.model.SowingEvent;
+import fi.metatavu.famifarm.persistence.model.WastageEvent;
+import fi.metatavu.famifarm.reporting.ReportException;
+import fi.metatavu.famifarm.rest.model.CellType;
+import fi.metatavu.famifarm.rest.model.EventType;
+import fi.metatavu.famifarm.rest.model.HarvestEventData.TypeEnum;
+
+/**
+ * Report for yield
+ * 
+ * @author Ville Koivukangas
+ */
+public class XlsxYieldReport extends AbstractXlsxReport {
+  @Inject
+  private LocalesController localesController;
+  
+  @Inject
+  private EventController eventController;
+  
+  @Inject
+  private WastageEventController wastageEventController;
+  
+  @Inject
+  private HarvestEventController harvestEventController;
+  
+  @Inject
+  private SowingEventController sowingEventController;
+  
+  @Inject
+  private BatchController batchController;
+  
+  @Inject
+  private LocalizedValueController localizedValueController;
+  
+  private long yield = 0;
```

Comment:
Move these private fields into create report method, otherwise these will be shared for all users accessing a report at the same time.